packages/graphql/src/entities/ResolvedConfig.ts

```diff
@@ -31,6 +31,19 @@ export abstract class ResolvedOptionBase {
   }
 }
 
+@nxs.objectType()
+export class NonNullResolvedStringOption extends ResolvedOptionBase {
```

Comment:
Pretty sure that we need to rethink the config layer types, this is feeling a bit clunky

Reply:
Yes agreed, I am not sure exactly how it should look, though. I think we need a better idea of all the cases before we change it - I think we should push this until it *doesn't* do something we need, then change it based on what we need it to do, rather than predicting what we think it should look like.

Reply:
My gut says it'd be easiest to just use a custom `JSON` scalar, from what I've seen in the designs, we're mostly going to be looking use this as a display in this sort of view:

```
type ConfigOption {
  field: String!
  source: ConfigOptionSourceEnum!
  value: JSON!
}
```

And then if we need better-typed individual fields, we could provide those fields directly in the schema

![image](https://user-images.githubusercontent.com/154748/133710924-a3127fa0-7063-43a5-9f53-cc4e748169fa.png)

Reply:
Duly noted, will look into this (not in this PR - separate ticket/PR).